server/controllers/user_controller.js

```diff
@@ -1,61 +1,132 @@
+        status: status.SERVER_ERROR,
+        error: 'server error',
+        e,
+      });
+    }
+  }
+
+  //  Signup
+  static signup = async (req, res) => {
+    try {
+      let {
+        first_name,
+        last_name,
+        email,
+        password,
+        is_admin,
+      } = req.body;
+
+      // Generate a unique ID
+      // Check if Email already exist
+      const user = await this.model().select('*', 'email=$1', [email]);
+      if (user[0]) {
+        // Catch any error if it rises
+        return res.status(status.REQUEST_CONFLICT).json({
+          status: status.REQUEST_CONFLICT,
+          error: `${email} already exists`,
 
-        return res.status(status.UNAUTHORIZED).send(user);
+        });
+      }
+      // Hash user password before being stored
+      password = await hashPassword.encryptPassword(password);
+      // console.log(password);
+      const cols = 'first_name, last_name, email, is_admin, password';
+      const sels = `'${first_name}', '${last_name}', '${email}', '${is_admin}', '${password}'`;
+      const rows = await this.model().insert(cols, sels);
+
+
+      let token = generateAuthToken(rows[0].user_id, rows[0].is_admin);
+
+      return res.status(status.RESOURCE_CREATED).json({
+        status: status.RESOURCE_CREATED,
+        message: 'User signed up successfully',
+        data: {
+          user_id: rows[0].user_id,
+          first_name: rows[0].first_name,
+          last_name: rows[0].last_name,
+          email: rows[0].email,
+          token,
+
+
+        },
+      });
+      // });
+    } catch (e) {
+      // Catch any error if it rises
+      return res.status(500).json({
+        status: status.SERVER_ERROR,
+        error: 'server error',
+        e,
+      });
+    }
+  }
+
+
+    static signin = async (req, res) => {
+      try {
+        const { email, password } = req.body;
+        const data = await this.model().select('*', 'email=$1', [email]);
+        //  If user credentials are correct
+        if (data[0] && hashPassword.decryptPassword(password, data[0].password)) {
+          const token = generateAuthToken(data[0].user_id, data[0].is_admin);
+          return res.status(status.REQUEST_SUCCEDED).json({
+            status: status.REQUEST_SUCCEDED,
+            message: 'user signed in successfully',
+            data: {
+              user_id: data[0].user_id,
+              first_name: data[0].first_name,
+              last_name: data[0].last_name,
+              email: data[0].email,
+              token,
+
+            },
+          });
+        }
+        // If no user is found with provided inputs
+        return res.status(status.UNAUTHORIZED).json({
+          status: status.UNAUTHORIZED,
+          error: 'Invalid Email or Password',
+        });
+      } catch (e) {
+        // Catch any error if it rises
+        return res.status(status.SERVER_ERROR).json({
+          status: status.SERVER_ERROR,
+          error: 'server error',
+          e,
```

Comment:
@victorkarangwa4 , you do not need to return this "e"

Reply:
Okay,
I actually use it in debugging and I forgot to remove it on the production environment, However, I have removed it.